media-source-respec.html

```diff
@@ -829,19 +829,17 @@
           <p>Follow these steps when <a def-id="duration"></a> needs to change to a <var>new duration</var>.</p>
           <ol>
             <li>If the current value of <a def-id="duration"></a> is equal to <var>new duration</var>, then return.</li>
-            <li>Set <var>old duration</var> to the current value of <a def-id="duration"></a>.
+			<li>If <var>new duration</var> is less than the highest start time reported by the <a def-id="buffered"></a> attribute across all <a>SourceBuffer</a> 
+			objects in <a def-id="sourceBuffers"></a>, then throw an <a def-id="invalid-state-error"></a> exception and abort these steps.</li>
+			  <p class="note">Duration reductions must first use <a def-id="remove"></a> to reduce the buffered range before updating <a def-id="duration"></a>.</p>
+            <li>Set <var>old duration</var> to the current value of <a def-id="duration"></a>.</li>
             <li>Update <a def-id="duration"></a> to <var>new duration</var>.</li>
-            <li>If the <var>new duration</var> is less than <var>old duration</var>, then run the
-              <a def-id="range-removal"></a> algorithm with <var>new duration</var> and
-              <var>old duration</var> as the start and end of the removal range.
-
-              <p class="note">This preserves audio frames and text cues that start before and end after the <a def-id="duration"></a>.</p></li>
-            <li>If a user agent is unable to partially render audio frames or text cues that start before and end after the <a def-id="duration"></a>, then run the following steps:
+            <li>If a user agent is unable to partially render audio frames or text cues that start before and end after the <a def-id="duration"></a>, then run the following steps:</li>
+			  <p class="note">This condition can occur because the <a def-id="coded-frame-removal-algorithm"></a> preserves audio frames and text cues that start before and end after the <a def-id="duration"></a>.</p>
```

Comment:
highest start time across all SourceBuffer.buffered could be time 0 in a continuously buffered media timeline. That would certainly allow duration reduction to truncate complete buffered coded frames (which start after the highest buffered range start time). So I don't think that solves this problem.
I also see a problem now that SourceBuffer.buffered is defined as intersection ranges, so there could be (in certain scenarios where a muxed media segment for a portion of the timeline has not been completely appended prior to an abort(), for example) complete coded frames buffered for one track buffer in a SourceBuffer, which start at or after the highest end time of the (intersection ranges) buffered attribute for that SourceBuffer. Duration reduction would thereby require those frame's removal (asynchronously), and such asynchronous removal as part of duration reduction was what this PR and set of bugs was trying to fix.
How about instead of forcing new duration to be >= the highest (start or end) time across all SourceBuffer.buffered, we instead: force new duration to be >= the highest end time across all track buffer ranges across all track buffers across all SourceBuffers? This would meet the intent of this PR more cleanly, IMO. And since we've recently clarified track buffer ranges normatively in the spec to be "treated as if it were stored in a normalized TimeRanges object", no ambiguous definition remains IIUC. This would also allow us to remove the "bumping up" of new duration in the scenario steps this comment is anchored on in the PR. WDYT?